src/commands/new/glee.ts

```diff
@@ -1,55 +1,139 @@
+    name: Flags.string({
+      char: 'n',
+      description: 'name of the project',
+      default: 'project',
+    }),
+    template: Flags.string({
+      char: 't',
+      description: 'name of the template',
+      default: 'default',
+    }),
+    file: Flags.string({
+      char: 'f',
+      description: 'path of the file',
+    }),
   };
 
   async run() {
     const { flags } = await this.parse(NewGlee); // NOSONAR
 
-    const projectName = flags.name;
-    const templateName = flags.template;
+    const { name: projectName, template: templateName, file } = flags;
 
     const PROJECT_DIRECTORY = join(process.cwd(), projectName);
-    const GLEE_TEMPLATES_DIRECTORY = resolve(__dirname, '../../../assets/create-glee-app/templates/', templateName);
-
-    try {
-      await fPromises.mkdir(PROJECT_DIRECTORY);
-    } catch (err: any) {
-      switch (err.code) {
-      case 'EEXIST':
-        this.error(`Unable to create the project. We tried to use "${projectName}" as the directory of your new project but it already exists (${PROJECT_DIRECTORY}). Please specify a different name for the new project. For example, run the following command instead:\n\n  asyncapi new ${this.commandName} --name ${projectName}-1\n`);
-        break;
-      case 'EACCES':
-        this.error(`Unable to create the project. We tried to access the "${PROJECT_DIRECTORY}" directory but it was not possible due to file access permissions. Please check the write permissions of your current working directory ("${process.cwd()}").`);
-        break;
-      case 'EPERM':
-        this.error(`Unable to create the project. We tried to create the "${PROJECT_DIRECTORY}" directory but the operation requires elevated privileges. Please check the privileges for your current user.`);
-        break;
-      default:
-        this.error(`Unable to create the project. Please check the following message for further info about the error:\n\n${err}`);
-      }
+
+    const GLEE_TEMPLATES_DIRECTORY = resolve(
+      __dirname,
+      '../../../assets/create-glee-app/templates/',
+      templateName
+    );
+
+    const currentTemplateName =
+      'https://github.com/KhudaDad414/glee-generator-template';
+
+    if (file && templateName && templateName !== 'default') {
+      this.error('You cannot use both --t and --f in the same command.');
     }
+    if (file) {
+      try {
+        const asyncapiInput = (await load(file)) || (await load());
+
+        const serversObject = asyncapiInput.toJson().servers;
+        const servers = Object.keys(serversObject);
+        const remoteServers = [];
+        for (const server of servers) {
+          const isRemote = await CliUx.ux.confirm(
+            `Is "${server}" a remote server`
+          );
+          remoteServers.push({ server, isRemote });
+        }
+
+        const selectedRemoteServers = remoteServers
+          .filter(({ isRemote }) => isRemote)
+          .map(({ server }) => server);
```

Comment:
maybe this can be simplified.
```suggestion
      const remoteServers = [];
        for (const server of servers) {
          if (await CliUx.ux.confirm(`Is "${server}" a remote server?`)) {
            remoteServers.push(server);
          }
        }
```

Reply:
used inquirer package.